doc/contributing/submitting-changes.chapter.md

```diff
@@ -82,6 +82,52 @@ If a security fix applies to both master and a stable release then, similar to r
 
 Critical security fixes may by-pass the staging branches and be delivered directly to release branches such as `master` and `release-*`.
 
+## Deprecating/removing packages {#submitting-changes-deprecating-packages}
+
+There is currently no policy when to remove a package.
+
+It should always be attempted to find a new maintainer or fix smaller issues first.
+
+### Steps to remove a package from nixpkgs
+
+We use jbidwatcher as an example for a discontinued project here.
+
+1. Have nixpkgs checked out locally and up to date.
+1. Create a new branch for your change, e.g. `git checkout -b jbidwatcher`
+1. Remove the actual package including it's directory, e.g. `rm -rf pkgs/applications/misc/jbidwatcher`
+1. Remove the package from list of all packages (`pkgs/top-level/all-packages.nix`).
+1. Add an alias for the package name in `pkgs/top-level/aliases.nix` (There is also `pkgs/misc/vim-plugins/aliases.nix`. Package sets typically do not have aliases, so we can't add them there.)
+
+    For example in this case:
+    ```
+    jbidwatcher = throw "jbidwatcher was discontinued in march 2021"; # added 2021-03-15
+    ```
+
+    The throw message should explain in short why the package was removed for users that still have it installed.
+
+1. Test if the changes introduced any issues with `nix-env -qaP -f . --show-trace`. It should show the list of packages and not an error.
+1. Commit the changes. Explain again why the package was removed. When it was declared discontinued upstream, add a link to the source.
+
+    ```
+    git add pkgs/applications/misc/jbidwatcher/default.nix pkgs/top-level/all-packages.nix pkgs/top-level/aliases.nix
+    git commit
+    ```
+
+    commit message:
+
+    ```
+    jbidwatcher: remove
+
+    project was discontinued in march 2021. the program does not work anymore because ebay changed the login.
+
+    https://web.archive.org/web/20210315205723/http://www.jbidwatcher.com/
+    ```
+
+1. Push changes to your GitHub fork with `git push`
```

Comment:
```suggestion
1. Push changes to your GitHub fork with `git push`.
```

Reply:
this would make it harder to select and copy
with the command on the right, you can just select from right

Reply:
Good point